servlet/src/main/java/io/undertow/servlet/core/BlockingWriterSenderImpl.java

```diff
@@ -183,14 +175,19 @@ private void performTransfer(FileChannel source, IoCallback callback) {
     @Override
     public void close(final IoCallback callback) {
         writer.close();
-        invokeOnComplete(callback);
+        if (writer.checkError()) {
+            callback.onException(exchange, this, new IOException());
+        } else {
+            invokeOnComplete(callback);
+        }
     }
 
     @Override
     public void close() {
         if(exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY).getDispatcherType() != DispatcherType.INCLUDE) {
             IoUtils.safeClose(writer);
         }
+        writer.checkError();
```

Comment:
@stuartwdouglas does it make sense to you to invoke checkError from close without the callback? Please, let me know what you think of this fix.

Reply:
Why are you calling it without checking the return value?

Reply:
@stuartwdouglas  That's the point. If I check the return value, there's no point, because there's no callback? Should I throw a RuntimeException? This is the part I'm lost at.